tests/test_config.py

```diff
@@ -93,6 +100,14 @@ def testTwoResource(self):
         self.assertEqual(res_config.minMemoryMB, 1024)
         self.assertEqual(res_config.minNumCores, 32)
 
+    def testMetadataDataset(self):
+        """Test for metadata dataset configuration.
+        """
+        config = pipeBase.PipelineTaskConfig()
+        self.assertTrue(config.saveMetadata)
+        config.saveMetadata = False
+        self.assertFalse(config.saveMetadata)
+
```

Comment:
So while I am all for unit tests, this test does not actually seem to be testing anything about code in pipe_base, but rather that pex_config works the way pex_config should, which presumably pex_config tests. If you wanted to test something about saveMetaData in a unit test, maybe just checking that the descriptor is present in the PipelineTaskConfig class, and or possibly its default value if you think changing it could cause future problems.

Reply:
I think about this test as testing for a default value of a parameter and that it can be changed to a different value. I can probably remove it altogether, this test initially was done for a different parameter (`metadataDataset`) with more complicated set of allowed values, it's not too useful for simple boolean option.